doc/rtd/development/packaging.rst

```diff
@@ -0,0 +1,105 @@
+.. _building_and_distribution:
+
+Building and Distribution
+*************************
+
+``Cloud-init`` is distributed as a system package and pre-installed in cloud
+images. Correct behavior depends on the tight integration of system services
+performing system-wide configuration during early boot stages. Because of
+this, cloud-init is not published to PyPI as it is not intended to be consumed
+as a pure-python package. Instead, cloud-init uses meson_ to build all
+dependent data files, system-initialization scripts and global configuration
+files.
+
+The build system used by ``cloud-init`` is `meson`_. ``Cloud-init`` is not
+published to pypi.org because it is not intended to be consumed directly as a
+pure-python library, SDK or API for managing cloud configuration.
+
+``Cloud-init`` correct behavior depends on the tight integration of
+system services performing :ref:`the early boot stages<boot_stages>` and
+system-wide configuration. Upstream uses `meson`_ build backend to support the
+distribution of all dependent data files, system-initialization scripts and
+global configuration files.
+
+Guidelines
+==========
+
+Build Dependencies
+------------------
+To use ``meson``, the following build dependencies should be available on the
+system:
+
+- ``python3``
+- ``meson >= 0.63.0``
+- ``pkgconf``
+- ``bash-completion``
+
+Additional dependencies for systemd environments:
+
+- ``systemd-devel``
+- ``udev``
+
+Local build procedure
+---------------------
+
+The custom ``init_system`` option affect paths where init scripts,
+configuration files, executables and data files are installed. The
+``init_system`` value defaults to ``systemd``.
+
+Other meson options used by this project are:
+
+- prefix
+- bindir
+- datadir
+- libexecdir
+- sysconfdir
+
+If the packaging system's meson option defaults are insufficient for your
+needs, they may be overridden with
+``meson setup -D<option_name>=<option_value>``.
+
+Steps to validate ``cloud-init`` package builds in a development environment:
+
+.. code-block:: bash
+
+   meson setup builddir -Dinit_system=systemd
```

Comment:
This option is configured as the default, so it shouldn't be necessary to set it.